### PR TITLE
Implement Picture for native

### PR DESCRIPTION
**Pull request by VeselovAlex**
Also introduce a solution for passing callbacks

### CODE REVIEW

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/Picture.js.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.jetbrains.skia.impl.NativePointer
+
+actual typealias AbortCallback = Int
+
+private external fun _registerCallback(cb: () -> Unit, data: Any?): AbortCallback
+
+internal actual fun registerAbortCallback(abort: (() -> Boolean)?): AbortCallback {
+    if (abort == null) { return 0 }
+
+    val data = arrayOf<Boolean?>(null) // We use array instead of structs due to field name mangling
```
Comment (eymar) [inline skiko/src/jsMain/kotlin/org/jetbrains/skia/Picture.js.kt:12]: @JsName annotation might help with field name mangling. But maybe using array for this is fine.

File: skiko/src/nativeMain/kotlin/org/jetbrains/skia/Actuals.native.kt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package org.jetbrains.skia
 
+import kotlinx.cinterop.staticCFunction
```
Comment (olonho) [inline skiko/src/nativeMain/kotlin/org/jetbrains/skia/Actuals.native.kt:3]: Is it used?

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/Picture.js.kt (hunk 1 above)
Comment (VeselovAlex) [reply skiko/src/jsMain/kotlin/org/jetbrains/skia/Picture.js.kt]: Fixed, thanks 🙏

File: skiko/src/nativeMain/kotlin/org/jetbrains/skia/Actuals.native.kt (hunk 2 above)
Comment (VeselovAlex) [reply skiko/src/nativeMain/kotlin/org/jetbrains/skia/Actuals.native.kt]: Fixed

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/impl/Native.js.kt [hunk 3]
```diff
@@ -194,13 +194,27 @@ actual class InteropScope actual constructor() {
         return toInterop(interopPointers.toIntArray())
     }
 
+    actual fun toInterop(callback: (() -> Boolean)?): InteropPointer {
+        if (callback == null) { return 0 }
+
+        val data = CallbackData<Boolean>(null)
+        return _registerCallback({ data.value = callback() }, data)
```
Comment (olonho) [inline skiko/src/jsMain/kotlin/org/jetbrains/skia/impl/Native.js.kt:201]: Where do we release callbacks?

Comment (VeselovAlex) [reply skiko/src/jsMain/kotlin/org/jetbrains/skia/impl/Native.js.kt:201]: Dispose is performed on Cpp side, see `common_interop.h#L396`

File: skiko/src/nativeJsMain/cpp/common_interop.cc [hunk 4]
```diff
@@ -369,3 +369,39 @@ namespace skija {
...
+}
+
+KBoolean callBooleanCallback(KInteropPointer cb) {
+    return callBooleanCallbackImpl(cb);
+}
+
+#else // __EMSCRIPTEN__
+
+void disposeCallback(KInteropPointer cb) {
+    EM_ASM({ _releaseCallback($0) }, cb);
```
Comment (VeselovAlex) [inline skiko/src/nativeJsMain/cpp/common_interop.cc:396]: Callbacks are disposed here

**Comment by olonho** [comment]
Needs rebase.

File: skiko/src/jvmMain/kotlin/org/jetbrains/skia/impl/Native.jvm.kt [hunk 5]
```diff
@@ -78,6 +79,7 @@ actual open class InteropScope actual constructor() {
     actual inline fun <reified T> InteropPointer.fromInterop(decoder: ArrayInteropDecoder<T>): Array<T> =
         this@fromInterop as Array<T>
     actual fun toInteropForArraysOfPointers(interopPointers: Array<InteropPointer>): InteropPointer = interopPointers
+    actual fun toInterop(callback: (() -> Boolean)?) = callback?.let { BooleanSupplier(it) as Any }
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skia/impl/Native.jvm.kt:82]: Not sure if this kind of conversion makes much sense, we could call `invoke()` on callback directly.

Comment (VeselovAlex) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skia/impl/Native.jvm.kt:82]: I've added it to make sure it is an instance of `BooleanSupplier` and has `getAsBoolean` for JNI

Comment (olonho) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skia/impl/Native.jvm.kt:82]: But why do we need BooleanSupplier here at all? Think it's an artefact of Java implementation. In Kotlin we have Function interface that does all we need.

Comment (VeselovAlex) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skia/impl/Native.jvm.kt]: Replaced with `kotlin.jvm.functions.Function0.invoke()`

File: skiko/src/jvmMain/cpp/common/interop.cc [hunk 6]
```diff
@@ -1202,3 +1200,36 @@ namespace skija {
...
+namespace kotlin {
+    namespace jvm {
+        namespace functions {
+            namespace Function0 {
+                jclass cls;
+                jmethodID invoke;
+
+                void onLoad(JNIEnv* env) {
+                    jclass local = env->FindClass("kotlin/jvm/functions/Function0");
+                    cls   = static_cast<jclass>(env->NewGlobalRef(local));
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/common/interop.cc:1213]: Extra space.

**Review by olonho — APPROVED** [review]

File: skiko/src/jvmMain/cpp/common/interop.cc (hunk 6 above)
Comment (VeselovAlex) [reply skiko/src/jvmMain/cpp/common/interop.cc]: Let's perform style fixed in a separate PR, there at least 3 places with same issue

**Pull request merged**